src/components/collectible-icon/collectible-icon.tsx

```diff
@@ -12,27 +12,38 @@ export const CollectibleIcon: FC<CollectibleIconProps> = ({ collectible, size })
   const [isLoaded, setIsLoaded] = useState(false);
   const [isLoadingFailed, setIsLoadingFailed] = useState(false);
 
-  return isDefined(collectible) && isDefined(collectible.thumbnailUri) ? (
+  if (!isDefined(collectible) || !isDefined(collectible.thumbnailUri) || !isDefined(collectible.artifactUri)) {
+    return (
+      <View
+        style={{
+          width: size,
+          height: size,
+          padding: formatSize(4)
+        }}
+      />
+    );
+  }
+
+  return (
     <View
       style={{
         width: size,
         height: size,
         padding: formatSize(4)
       }}
     >
-      {isDefined(collectible.artifactUri) ? (
```

Comment:
why don't just extend this condition? 
The same but with fewer changed lines, isn't it?
<img width="930" alt="image" src="https://user-images.githubusercontent.com/10281464/161059826-fe7594d2-9238-4031-9276-3d9026b2965b.png">